src/Apps/Order/Components/OfferInput.tsx

```diff
@@ -20,28 +20,28 @@ export const OfferInput: FC<OfferInputProps> = ({
   onChange,
   value,
 }) => {
+  const formatValueForDisplay = (val: number | undefined) => {
+    if (val !== undefined) {
+      return val.toLocaleString()
+    }
+    return ""
+  }
+
   return (
     <Input
       id={id}
       title={noTitle ? "" : "Your offer"}
       type="text"
       pattern="[0-9]"
       error={showError ? "Offer amount missing or invalid." : false}
+      inputMode={"numeric"}
       onFocus={onFocus}
       onBlur={onBlur}
-      value={value}
+      value={formatValueForDisplay(value)}
       onChange={ev => {
         const currentValue = ev.currentTarget.value
-        const nonDigitMatch = currentValue.match(/\D/)
-
-        if (nonDigitMatch) {
-          const cursorOffset = currentValue.indexOf(nonDigitMatch[0])
```

Comment:
having the commas interferes with this logic to keep the cursor in the same place when inputting non numbers. I think we can live without it, specially if we have the number keyboard for mobile I'm introducing in the PR as well